server/webhook.go

```diff
@@ -213,78 +214,71 @@ func (s *Server) processPushEvent(ctx context.Context, repositoryList []*api.Rep
-				zap.String("file", item.FileName),
-				zap.Error(err),
-			)
-			continue
-		}
+	var createdMessageList []string
+	repoID2FileItemList := groupFileInfoByRepo(distinctFileList, repositoryList)
+	for _, fileInfoList := range repoID2FileItemList {
+		repository := fileInfoList[0].repository
 		pushEvent := baseVCSPushEvent
 		pushEvent.VCSType = repository.VCS.Type
 		pushEvent.BaseDirectory = repository.BaseDirectory
-		pushEvent.FileCommit = vcs.FileCommit{
-			ID:          item.Commit.ID,
-			Title:       item.Commit.Title,
-			Message:     item.Commit.Message,
-			CreatedTs:   item.Commit.CreatedTs,
-			URL:         item.Commit.URL,
-			AuthorName:  item.Commit.AuthorName,
-			AuthorEmail: item.Commit.AuthorEmail,
-			Added:       item.FileName,
-		}
-
-		var createdMessageList []string
-		repoID2ActivityCreateList := make(map[int][]*api.ActivityCreate)
-		createdMessage, created, activityCreateList, err := s.processFile(
+		createdMessage, created, activityCreateList, err := s.processFilesInProject(
 			ctx,
 			pushEvent,
 			repository,
-			item.FileName,
-			item.ItemType,
-			migrationInfo,
-			fType,
+			fileInfoList,
 		)
 		if err != nil {
 			return nil, err
 		}
 		if created {
 			createdMessageList = append(createdMessageList, createdMessage)
-		}
-		repoID2ActivityCreateList[repository.ID] = append(repoID2ActivityCreateList[repository.ID], activityCreateList...)
-
-		if len(createdMessageList) == 0 {
-			for _, repo := range repositoryList {
-				if activityCreateList, ok := repoID2ActivityCreateList[repo.ID]; ok {
-					for _, activityCreate := range activityCreateList {
-						if _, err := s.ActivityManager.CreateActivity(ctx, activityCreate, &ActivityMeta{}); err != nil {
-							log.Warn("Failed to create project activity for the ignored repository file",
-								zap.Error(err),
-							)
-						}
-					}
+		} else {
+			for _, activityCreate := range activityCreateList {
+				if _, err := s.ActivityManager.CreateActivity(ctx, activityCreate, &ActivityMeta{}); err != nil {
+					log.Warn("Failed to create project activity for the ignored repository files", zap.Error(err))
 				}
 			}
 		}
-		createdMessages = append(createdMessages, createdMessageList...)
 	}
 
-	if len(createdMessages) == 0 {
+	if len(createdMessageList) == 0 {
 		var repoURLs []string
 		for _, repo := range repositoryList {
 			repoURLs = append(repoURLs, repo.WebURL)
 		}
-		log.Warn("Ignored push event because no applicable file found in the commit list", zap.Any("repos", repoURLs))
+		log.Warn("Ignored push event because no applicable file found in the commit list", zap.Strings("repos", repoURLs))
+	}
+
+	return createdMessageList, nil
+}
+
+type fileInfo struct {
+	item          vcs.DistinctFileItem
+	migrationInfo *db.MigrationInfo
+	fType         fileType
+	repository    *api.Repository
+}
+
+func groupFileInfoByRepo(distinctFileList []vcs.DistinctFileItem, repositoryList []*api.Repository) map[int][]fileInfo {
```

Comment:
Could you provide docstring for what this function is doing?

Reply:
Done.